### ISSUE TITLE
[Bug]: DATA RACE during bvt

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#6852]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): b13440ebc66471642119070c503f652960f14610
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

full log: https://github.com/sukki37/matrixone/suites/9443470503/artifacts/446692749

```
2022/11/22 12:08:23.423334 +0000 DEBUG hakeeper-client-backend morpc/backend.go:557 read response {"remote": "127.0.0.1:32001", "backend-id": "381f67cc-bf9f-43aa-8f02-2f5d3343196e", "request-id": 2, "response": ""}
2022/11/22 12:08:23.423449 +0000 INFO cnservice/server.go:214 Shutdown The Server With Ctrl+C | Ctrl+\.
==================
WARNING: DATA RACE
Write at 0x000006ffed70 by goroutine 1410:
  github.com/matrixorigin/matrixone/pkg/common/mpool.InitCap()
      /home/runner/work/matrixone/matrixone/head/pkg/common/mpool/mpool.go:453 +0x148
  github.com/matrixorigin/matrixone/pkg/cnservice.(*service).initMOServer()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/server.go:218 +0xda
  github.com/matrixorigin/matrixone/pkg/cnservice.NewService()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/server.go:100 +0x9c4
  main.startCNService.func1()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:152 +0x33a
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:258 +0xd1

Previous read at 0x000006ffed70 by goroutine 1393:
  github.com/matrixorigin/matrixone/pkg/common/mpool.GlobalCap()
      /home/runner/work/matrixone/matrixone/head/pkg/common/mpool/mpool.go:461 +0x215
  github.com/matrixorigin/matrixone/pkg/common/mpool.(*MPool).Alloc()
      /home/runner/work/matrixone/matrixone/head/pkg/common/mpool/mpool.go:524 +0x22c
  github.com/matrixorigin/matrixone/pkg/container/vector.(*Vector).extend()
      /home/runner/work/matrixone/matrixone/head/pkg/container/vector/tools.go:327 +0x265
  github.com/matrixorigin/matrixone/pkg/container/vector.appendOne[...]()
      /home/runner/work/matrixone/matrixone/head/pkg/container/vector/vector.go:711 +0x6b
  github.com/matrixorigin/matrixone/pkg/container/vector.(*Vector).Append()
      /home/runner/work/matrixone/matrixone/head/pkg/container/vector/vector.go:780 +0xc58
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.genInsertBatch()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/tools.go:1042 +0x63d
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*DB).init()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/db.go:218 +0x1a04
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.New()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/engine.go:51 +0x38e
  github.com/matrixorigin/matrixone/pkg/cnservice.(*service).initDistributedTAE()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/distributed_tae.go:58 +0x21b
  github.com/matrixorigin/matrixone/pkg/cnservice.(*service).initEngine()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/server.go:246 +0x212
  github.com/matrixorigin/matrixone/pkg/cnservice.(*service).initMOServer()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/server.go:222 +0x224
  github.com/matrixorigin/matrixone/pkg/cnservice.NewService()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/server.go:100 +0x9c4
  main.startCNService.func1()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:152 +0x33a
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:258 +0xd1

Goroutine 1410 (running) created at:
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:253 +0x16a
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).RunNamedTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:169 +0x176
  main.startCNService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:150 +0x1b2
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:132 +0x1a4
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:112 +0x339
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:51 +0x188
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:81 +0x32b

Goroutine 1393 (running) created at:
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:253 +0x16a
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).RunNamedTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:169 +0x176
  main.startCNService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:150 +0x1b2
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:132 +0x1a4
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:112 +0x339
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:51 +0x188
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:81 +0x32b
==================
2022/11/22 12:08:23.425344 +0000 INFO cnservice/server.go:221 Initialize the engine ...
2022/11/22 12:08:23.427730 +0000 INFO frontend/server.go:102 start server on unix domain socket /var/lib/mysql/mysql.sock failed with listen unix /var/lib/mysql/mysql.sock: bind: permission denied
2022/11/22 12:08:23.427960 +0000 INFO frontend/server.go:40 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428014 +0000 INFO frontend/server.go:41 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428054 +0000 INFO frontend/server.go:42 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428093 +0000 INFO frontend/server.go:43 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428141 +0000 INFO frontend/server.go:44 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428183 +0000 INFO frontend/server.go:45 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428219 +0000 INFO frontend/server.go:46 Server Listening on : 0.0.0.0:16001 
2022/11/22 12:08:23.428259 +0000 INFO frontend/server.go:50 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428291 +0000 INFO frontend/server.go:51 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428324 +0000 INFO frontend/server.go:52 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428358 +0000 INFO frontend/server.go:53 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428387 +0000 INFO frontend/server.go:54 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428422 +0000 INFO frontend/server.go:55 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.428458 +0000 DEBUG v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:287 application accept loop started {"listen-addresses": "[0.0.0.0:16001]"}
2022/11/22 12:08:23.428514 +0000 DEBUG v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:289 application accept loop stopped {"listen-addresses": "[0.0.0.0:16001]"}
2022/11/22 12:08:23.428561 +0000 DEBUG v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:224 application started {"listen-addresses": "[0.0.0.0:16001]"}
2022/11/22 12:08:23.428629 +0000 DEBUG cn-service v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:287 application accept loop started {"uuid": "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "cn-server", "listen-addresses": "[0.0.0.0:7001]"}
2022/11/22 12:08:23.428677 +0000 DEBUG cn-service v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:289 application accept loop stopped {"uuid": "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "cn-server", "listen-addresses": "[0.0.0.0:7001]"}
2022/11/22 12:08:23.428716 +0000 DEBUG cn-service v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:224 application started {"uuid": "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf", "name": "cn-server", "listen-addresses": "[0.0.0.0:7001]"}
{"level":"INFO","time":"2022/11/22 12:08:23.428872 +0000","name":"rpc-client[cn-client]","caller":"morpc/client.go:328","msg":"gc inactive backends task started"}
2022/11/22 12:08:23.428999 +0000 INFO cn-service logutil/action.go:49 start to run async task {"uuid": "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf", "action": "run-async-task", "async-task": "cnservice/wait-system-init-task"}
2022/11/22 12:08:23.429073 +0000 DEBUG cn-service cnservice/server_task.go:144 wait all init task completed task started {"uuid": "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf"}
2022/11/22 12:08:23.429166 +0000 INFO cn-service logutil/action.go:49 start to run async task {"uuid": "dd1dccb4-4d3c-41f8-b482-5251dc7a41bf", "action": "run-async-task", "async-task": "cnservice/heartbeat-task"}
2022/11/22 12:08:23.429390 +0000 INFO cn-service.rpc-client[txn-rpc-sender] morpc/client.go:328 gc inactive backends task started {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf"}
2022/11/22 12:08:23.430890 +0000 INFO cn-service.rpc-client[txn-rpc-sender] morpc/client.go:302 gc idle backends task started {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf"}
2022/11/22 12:08:23.443166 +0000 INFO frontend/server.go:102 start server on unix domain socket /var/lib/mysql/mysql.sock failed with listen unix /var/lib/mysql/mysql.sock: bind: permission denied
2022/11/22 12:08:23.443488 +0000 INFO frontend/server.go:40 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443552 +0000 INFO frontend/server.go:41 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443595 +0000 INFO frontend/server.go:42 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443629 +0000 INFO frontend/server.go:43 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443609 +0000 INFO cn-service logutil/action.go:49 start to run async task {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf", "action": "run-async-task", "async-task": "cnservice/wait-system-init-task"}
2022/11/22 12:08:23.443720 +0000 DEBUG cn-service cnservice/server_task.go:144 wait all init task completed task started {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf"}
2022/11/22 12:08:23.443666 +0000 INFO frontend/server.go:44 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443810 +0000 INFO frontend/server.go:45 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443853 +0000 INFO frontend/server.go:46 Server Listening on : 0.0.0.0:16002 
2022/11/22 12:08:23.443905 +0000 INFO frontend/server.go:50 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443948 +0000 INFO frontend/server.go:51 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.443982 +0000 INFO frontend/server.go:52 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.444020 +0000 INFO frontend/server.go:53 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.444056 +0000 INFO frontend/server.go:54 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.444090 +0000 INFO frontend/server.go:55 ++++++++++++++++++++++++++++++++++++++++++++++++
2022/11/22 12:08:23.444126 +0000 DEBUG v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:287 application accept loop started {"listen-addresses": "[0.0.0.0:16002]"}
2022/11/22 12:08:23.444174 +0000 DEBUG v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:289 application accept loop stopped {"listen-addresses": "[0.0.0.0:16002]"}
2022/11/22 12:08:23.444216 +0000 DEBUG v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:224 application started {"listen-addresses": "[0.0.0.0:16002]"}
2022/11/22 12:08:23.444275 +0000 DEBUG cn-service v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:287 application accept loop started {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf", "name": "cn-server", "listen-addresses": "[0.0.0.0:7002]"}
2022/11/22 12:08:23.444326 +0000 DEBUG cn-service v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:289 application accept loop stopped {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf", "name": "cn-server", "listen-addresses": "[0.0.0.0:7002]"}
2022/11/22 12:08:23.444365 +0000 DEBUG cn-service v2@v2.0.3-0.20221121133015-81355a910e4a/application.go:224 application started {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf", "name": "cn-server", "listen-addresses": "[0.0.0.0:7002]"}
==================
WARNING: DATA RACE
Write at 0x00000608f330 by goroutine 1410:
  github.com/matrixorigin/matrixone/pkg/cnservice/cnclient.NewCNClient()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/cnclient/client.go:93 +0x1e9
  main.startCNService.func1()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:166 +0x3ac
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:258 +0xd1

Previous read at 0x00000608f330 by goroutine 1393:
  github.com/matrixorigin/matrixone/pkg/cnservice/cnclient.NewCNClient()
      /home/runner/work/matrixone/matrixone/head/pkg/cnservice/cnclient/client.go:109 +0x96e
  main.startCNService.func1()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:166 +0x3ac
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:258 +0xd1

Goroutine 1410 (running) created at:
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:253 +0x16a
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).RunNamedTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:169 +0x176
  main.startCNService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:150 +0x1b2
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:132 +0x1a4
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:112 +0x339
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:51 +0x188
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:81 +0x32b

Goroutine 1393 (running) created at:
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:253 +0x16a
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).RunNamedTask()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:169 +0x176
  main.startCNService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:150 +0x1b2
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:132 +0x1a4
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:112 +0x339
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:51 +0x188
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:81 +0x32b
==================
2022/11/22 12:08:23.444776 +0000 INFO cn-service logutil/action.go:49 start to run async task {"uuid": "dd1dccb5-4d3c-41f8-b482-5251dc7a41bf", "action": "run-async-task", "async-task": "cnservice/heartbeat-task"}
{"level":"INFO","time":"2022/11/22 12:08:23.444827 +0000","name":"rpc-client[cn-client]","caller":"morpc/client.go:328","msg":"gc inactive backends task started"}

```

### Steps to Reproduce

```Markdown
run multi-cn bvt with race enabled
```

**Comment by nnsgmsone** [comment]
mpool data trace fixed by #6866

**Comment by zhangxu19830126** [comment]
fixed in #6873  and #6881

**Comment by sukki37** [comment]
Not repro
